tests/testdata/qgis_server/getcapabilities.txt

```diff
@@ -199,6 +199,54 @@ Content-Type: text/xml; charset=utf-8
      </LegendURL>
     </Style>
    </Layer>
+   <Layer queryable="1">
+    <Name>&lt;test layer name></Name>
```

Comment:
Is this the expected output? Shouldn't it be 
```suggestion
    <Name>&lt;test layer name&gt;</Name>
```
??

Reply:
> Is this the expected output?

In the scope of this PR, yes, because this is from the GetCapabilities output which isn't affected by `convertFeatureInfoToHtml()`.

This is also expected with `master` -> replacing only `<` with `&lt;` (and not `>` with `&gt;`) is the case for the XML when you save a QGIS project.
This is also where these changes from the GetCapabilities output are coming from, there are now two additional layers with tag chars in name/title included for `test_project.qgs` for testing `convertFeatureInfoToHtml()`.

Maybe it's enough to just replace the opening tag char for XML - since [toHtmlEscaped()](https://doc.qt.io/qt-5/qstring.html#toHtmlEscaped) is available, I've opted to use it for `convertFeatureInfoToHtml()` and thus there is now also the closing tag char replaced (and also `&` and `"`).